### DISCUSSION TITLE
Best practices | Correct way to implement firebase/firestore sdk

### DISCUSSION THREAD

**Discussion by ithustle** [invertase/react-native-firebase#5594] [General]
Hi there.

What is the correct way or best practices when implement firebase sdk to use firestore?

Maybe is a dumb question but I'm facing some issues when I try to save an information on a collection of my firestore database. Sometimes works when I try to save an info, however in some cases this operation only works if I close my app (killing the process) and then try it again.  It only happens using firebase sdk, if I do that through cloud functions it works fine. I'm using firebase sdk to get realtime feature on my app.

A snippet code below how I do that:

````
export const playlistsRef = firebase.firestore().collection("playlists");
...

export const createPlaylist = async data => {
    const user = getUserProfile();
    const documentId = `${user.phoneNumber || user.email}_${data.playlistName}`;
    await playlistsRef
        .doc(documentId)
        .set(data);
}
````

Any help, tips ... would be thankful.

**Comment by mikehardy** [comment]
For general questions I generally refer people to stackoverflow, where the firebase community in general does collaborate (including the firebase employees). But you may also like to simply use our e2e tests as a reference, by design they aim for as close to 100% coverage of firestore capabilities so whatever you want to do, there should be an example in there...

https://github.com/invertase/react-native-firebase/tree/master/packages/firestore/e2e

**Reply by ithustle** [reply]
Thanks, @mikehardy. I'll double check it.
